### ISSUE TITLE
tf.estimator generator_input_fn no padding queue?

### ISSUE THREAD

**Issue by Scitator** [tensorflow/tensorflow#10680]
Hello again! 
While I was examining TF source code for [10597 issue](https://github.com/tensorflow/tensorflow/issues/10597) solution, I found another bug or lack of important feature, such as `PaddingFIFOQueue` for all of `tf.estimator` input pipelines. Such option is very important for seq2seq tasks with dynamic shapes.

As a solution I found a hacky way to generate `[1, batch_size, time]` batches and squeeze them to `[batch_size, time]` in the model. Source code: [tensorflow](https://github.com/Scitator/tensorflow), [model](https://github.com/Scitator/TF-seq2seq/tree/tf_master_seq2seq), [notebook](https://gist.github.com/Scitator/d72cef607d23200074dfbbc1cbce3b55).

As you can understand, I am not very happy with such solution, that's why I am asking for help to solve it in more correct and tensorflow way. Any suggestions?

**Comment by aselle** [comment]
Can you state the problem more clearly? What would the feature be? How would it work? Why is it needed?

**Comment by Scitator** [comment]
The problem:
typically, `generator_input_fn` expects that python generator yields numpy array of features with same size for each training example. **But** for some cases, for example, seq2seq models, each training example has different size (just different lens, really). Dynamic shapes broke up current input pipeline:
1. firstly, in `_enqueue_data` function: it use first yielded example for shape specification and remember it. For dynamic shapes it is completely wrong, cause...you know, shape are dynamic. [source](https://github.com/Scitator/tensorflow/blob/a827151b7a6ac152d63e2241a3d7a1d1aaf15aff/tensorflow/python/estimator/inputs/queues/feeding_functions.py#L365)
2. secondly, at feed stage, when it converts list of numpy array to matrix: if shape are equal, all works correctly, but our shapes are not, so -> exception. [source](https://github.com/Scitator/tensorflow/blob/master/tensorflow/python/client/session.py#L1075)

Why is it needed?
Personally, the main reason why it needed - give simplified input pipeline interface for seq2seq models. It can be done through `PaddingFIFOQueue` and `placeholder_with_default`, but for `tf.estimator` usage we still need some input pipeline function

What would the feature be? How would it work?
The feature is quite simple, it could be an additional flag, or `shuffle` replacement to allow user to use  `PaddingFIFOQueue` under the hood for batch preprocessing. 

For now, I clearly understand how to do it for generator pipeline, although for other ones - dont think so. As for me, only generator one can handle dynamic shapes, other one, such as numpy source or pandas daaframe already want data to be same size.

**Comment by aselle** [comment]
Thank you your new description is much more clear!

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and this issue has an assignee.Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tfboyd: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tfboyd: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tfboyd: It has been 343 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tfboyd: It has been 358 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.
